### PR TITLE
Add support for editing and deleting your last Twt

### LINKED ISSUE #85: Twt deletion
I think it would be nice to be able to [correct mistakes](https://twtxt.net/user/lukem#7c7sz6cxst7jrelw5brfhjt25mrsffypbzsh7y42z2worivcgfca) in latest twt, at least for a limited time, either with an 'Edit' button or by deleting the twt and posting it again. Sausage fingers and poorly calibrated autocorrections would be happier. :)

### PR DESCRIPTION
**Pull request by prologic**
Closes #85

### CODE REVIEW

File: internal/static/js/twtxt.js [hunk 1]
```diff
@@ -67,13 +67,63 @@ function reply(e) {
...
+
+  text.setSelectionRange(size, size);
+
+  u("#tweetForm").attr("method", "PATCH");
+}
+
+function deleteTweet(e) {
+  e.preventDefault();
+
+  if (confirm("Are you sure you want to delete this tweet? This cannot be undone!")) {
```
Comment (aaadonai) [inline internal/static/js/twtxt.js:92]: Should it say twt instead of tweet?

File: internal/handlers.go [hunk 2]
```diff
@@ -443,6 +443,22 @@ func (s *Server) PostHandler() httprouter.Handle {
 
 		postas := strings.ToLower(strings.TrimSpace(r.FormValue("postas")))
 
+		// TODO: Support deleting/patching last feed (`postas`) tweet too.
+		if r.Method == http.MethodDelete || r.Method == http.MethodPatch {
+			if err := DeleteLastTweet(s.config, ctx.User); err != nil {
+				ctx.Error = true
+				ctx.Message = "Error deleting last tweet"
```
Comment (aaadonai) [inline internal/handlers.go:450]: Also here: Should it say twt instead of tweet?

File: internal/handlers.go [hunk 3]
```diff
@@ -443,6 +443,22 @@ func (s *Server) PostHandler() httprouter.Handle {
...
+				s.render("error", w, ctx)
+			}
+
+			if r.Method != http.MethodDelete {
+				ctx.Error = false
+				ctx.Message = "Successfully deleted last tweet"
```
Comment (aaadonai) [inline internal/handlers.go:456]: Ops. Found another one. twt instead of tweet.

File: internal/tweets.go [hunk 4]
```diff
@@ -21,6 +23,10 @@ const (
 	feedsDir = "feeds"
 )
 
+var (
+	ErrInvalidTweetLine = errors.New("error: invalid tweet line parsed")
```
Comment (aaadonai) [inline internal/tweets.go:27]: And another one. :-)

File: internal/tweets.go [hunk 5]
```diff
@@ -117,32 +123,57 @@ func ExpandMentions(conf *Config, db Store, user *User, text string) string {
...
 func AppendSpecial(conf *Config, db Store, specialUsername, text string) error {
 	user := &User{Username: specialUsername}
 	user.Following = make(map[string]string)
 	return AppendTweet(conf, db, user, text)
 }
 
 func AppendTweet(conf *Config, db Store, user *User, text string) error {
+	text = strings.TrimSpace(text)
+	if text == "" {
+		return fmt.Errorf("cowardly refusing to tweet empty text, or only spaces")
```
Comment (aaadonai) [inline internal/tweets.go:158]: this is also using tweet but it seems like it is only a log message?

File: internal/static/js/twtxt.js [hunk 6]
```diff
@@ -67,13 +67,63 @@ function reply(e) {
...
+    u("#post").html("<i class=\"icss-spinner icss-pulse\"></i>&nbsp;Deleting...");
+    u("#post").attr("disabled", true);
+    Twix.ajax({
+      type: "DELETE",
+      url: u("#tweetForm").attr("action"),
+      success: function(data) {
```
Comment (dooven) [inline internal/static/js/twtxt.js:98]: Should we add an window.alert, if an errors occurs?

File: internal/static/js/twtxt.js (hunk 1 above)
Comment (prologic) [reply internal/static/js/twtxt.js:92]: Yes! I've been meaning to do a mass sed on the codebase ;)

File: internal/handlers.go (hunk 2 above)
Comment (prologic) [reply internal/handlers.go:450]: Yup 👍

File: internal/tweets.go (hunk 5 above)
Comment (prologic) [reply internal/tweets.go:158]: Yeah

**Comment by prologic** [comment]
@aaadonai If you see on other obvious bugs/problems I'll mere this and so a mass code refactor; effectively `s/Tweet/Twt/g`, `s/tweet/twt/g`, `s/Tweeter/Poster`

**Pull request merged**